### PR TITLE
Add 1 test case for Convtranspose1D in op microbenchmark

### PR DESCRIPTION
**Pull request by DiweiSun**
Operator Convtransposd1d suffers performance regression with specific shape, #120982. Then we'd like to have this shape included into op level benchmark in this PR. 

I reproduced the regression that convtranspos1d with shape [2016, 1026, 1024, 256, 1, 224]. Here is the summary:

Hardware info: Intel SPR8480-56cores per socket with frequency=2.1G.
Performance comparison between torch 1.13 vs. torch 2.2
Benchmarking **PyTorch1.13**: ConvTranspose1d Mode: Eager
Name: ConvTranspose1d_IC2016_OC1026_kernel1024_stride256_N1_L224_cpu
Input: IC: 2016, OC: 1026, kernel: 1024, stride: 256, N: 1, L: 224, device: cpu
Forward Execution Time (s) : **0.96s**

Benchmarking **PyTorch2.2:** ConvTranspose1d
Mode: Eager
Name: ConvTranspose1d_IC2016_OC1026_kernel1024_stride256_N1_L224_cpu
Input: IC: 2016, OC: 1026, kernel: 1024, stride: 256, N: 1, L: 224, device: cpu
Forward Execution Time (s) : **7.988s**

Also benchmarking for 7 rounds to check the variance. 

  | Round1 | Round2 | Round3 | Round4 | Round5 | Round6 | Round7 | Normalized   Variance
-- | -- | -- | -- | -- | -- | -- | -- | --
Pytorch1.13 | 0.971 | 0.972 | 0.969 | 0.970 | 0.972 | 0.970 | 0.971 | 0.0002%
Pytorch 2.2 | 8.064 | 8.053 | 8.027 | 7.927 | 7.971 | 7.929 | 7.902 | 0.0059%
Ratio v2.2 vs.   v1.13(Lower is better) | 8.31 | 8.28 | 8.29 | 8.18 | 8.20 | 8.18 | 8.14 |  

Reproduce script：
numctl -N 0 python -m pt.conv_test

### CODE REVIEW

**Review by chuanqi129 — COMMENTED** [review]
Todo:

- [x] Add background issue https://github.com/pytorch/pytorch/issues/120982 introduce in description
- [x] Verify the performance stability of this microbenchmark tool on CPU
- [x] Collect performance data for this new test case based on current pytorch version and original version with performance issue https://github.com/pytorch/pytorch/issues/120982
- [x] Summary current test cases in this microbenchmark [nice to have]

**Comment by DiweiSun** [comment]
Update the current status of OP benchmarks:
<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/diweisun/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/diweisun/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<body link="#467886" vlink="#96607D">


OP lists | Forward | Backward | Total Cases
-- | -- | -- | --
abs | 10 |   | 10
acos | 10 |   | 10
AdaptiveMaxPool2d | 17 |   | 17
AdaptiveMaxPool3d | 17 |   | 17
add | 71 | 33 | 104
add_relu | 24 |   | 24
add_scalar | 24 |   | 24
addbmm | 16 | 64 | 80
addmm | 11 | 44 | 55
addr | 8 | 32 | 40
argsort | 5 |   | 5
as_strided | 7 |   | 7
asin | 10 |   | 10
atan | 10 |   | 10
AvgPool1d | 9 |   | 9
AvgPool2d | 17 |   | 17
AvgPool3d | 17 |   | 17
batchnorm | 20 | 40 | 60
bernoulli | 5 |   | 5
bmm | 5 |   | 5
bmm(actualop=bmm | 16 |   | 16
bmm(actualop=matmul | 16 |   | 16
cat | 25 |   | 25
cauchy | 5 |   | 5
ceil | 10 |   | 10
channel_shuffle | 76 |   | 76
chunk | 11 |   | 11
clone | 5 |   | 5
Conv1d | 10 |   | 10
Conv2d | 9 |   | 9
Conv2dPointwise | 9 |   | 9
Conv3d | 1 |   | 1
ConvTranspose1d | 21 |   | 21
ConvTranspose2d | 9 |   | 9
ConvTranspose3d | 1 |   | 1
copy | 35 |   | 35
cos | 10 |   | 10
cosh | 5 |   | 5
DequantizePerChannel | 109 |   | 109
DequantizePerTensor | 37 |   | 37
diag | 3 |   | 3
digamma | 10 |   | 10
einsum_bmm | 5 |   | 5
einsum_elementwis | 30 |   | 30
embedding | 24 |   | 24
embeddingbag | 96 |   | 96
eq | 48 |   | 48
erf | 10 |   | 10
erfc | 10 |   | 10
erfinv | 5 |   | 5
exp | 10 |   | 10
expm1 | 10 |   | 10
exponential | 5 |   | 5
FakeQuantize | 17 |   | 17
FloatToHalfTensorConversionBenchmark | 25 |   | 25
floor | 10 |   | 10
fmod | 33 |   | 33
frac | 10 |   | 10
FractionalMaxPool2d | 17 |   | 17
FractionalMaxPool3d | 17 |   | 17
functional.celu | 52 |   | 52
functional.elu | 52 |   | 52
functional.hardsigmoid | 52 |   | 52
functional.hardswish | 52 |   | 52
functional.hardtanh | 52 |   | 52
functional.leaky | 52 |   | 52
functional.sigmoid | 52 |   | 52
functional.tanh | 52 |   | 52
gather | 10 |   | 10
ge | 48 |   | 48
GeluBenchmark | 8 |   | 8
GroupNormBenchmark | 4 |   | 4
gt | 48 |   | 48
half | 5 |   | 5
HalfToFloatTensorConversionBenchmark | 25 |   | 25
hardshrink | 5 |   | 5
Hardsigmoid | 10 |   | 10
Hardswish | 10 |   | 10
HistogramObserver | 26 |   | 26
HistogramObserverCalculateQparams | 26 |   | 26
index_select | 14 |   | 14
InstanceNormBenchmark | 2 |   | 2
interpolate | 116 |   | 116
LayerNormBenchmark | 4 |   | 4
le | 48 |   | 48
learnable | 68 | 272 | 340
lgamma | 5 |   | 5
linear | 11 |   | 11
linear_prepack | 11 |   | 11
linear_unpack | 11 |   | 11
log | 10 |   | 10
log10 | 10 |   | 10
log1p | 10 |   | 10
log2 | 10 |   | 10
logit | 10 |   | 10
LogSoftmax | 24 |   | 24
long | 5 |   | 5
lt | 48 |   | 48
matmul | 11 |   | 11
MaxPool1d | 9 |   | 9
MaxPool2d | 17 |   | 17
MaxPool3d | 17 |   | 17
MinMaxObserver | 18 |   | 18
MovingAverageMinMaxObserver | 18 |   | 18
MovingAveragePerChannelMinMaxObserver | 18 |   | 18
mul | 54 |   | 54
mul_scalar | 24 |   | 24
nan_to_num | 104 |   | 104
ne | 48 |   | 48
neg | 10 |   | 10
normal | 5 |   | 5
original | 100 | 204 | 304
PerChannelMinMaxObserver | 18 |   | 18
q_argsort | 13 |   | 13
q_clone | 13 |   | 13
q_interpolate | 23 |   | 23
q_mean | 13 |   | 13
q_relu | 26 |   | 26
q_sort | 13 |   | 13
QAdaptiveAvgPool2dBenchmark | 83 |   | 83
qatEmbedding_num_embeddings | 24 |   | 24
qatEmbeddingBag_embeddingbags | 48 |   | 48
QAvgPool2dBenchmark | 19 |   | 19
QBatchNorm1d | 1 |   | 1
QBatchNorm2d | 1 |   | 1
qcat | 162 |   | 162
QConv1d | 10 |   | 10
QConv2d | 9 |   | 9
QDynamicLinear | 11 |   | 11
qembeddingbag_byte_prepack_num_embeddings | 162 |   | 162
qEmbeddingBag_embeddingbags | 48 |   | 48
QGroupNormBenchmark | 4 |   | 4
QInstanceNormBenchmark | 2 |   | 2
QLayerNormBenchmark | 4 |   | 4
QLinear | 11 |   | 11
QLSTM | 4 |   | 4
QMaxPool2dBenchmark | 19 |   | 19
QMethodTensorInputCopyBenchmark | 18 |   | 18
qtopk | 37 |   | 37
QuantizePerChannel | 109 |   | 109
QuantizePerTensor | 37 |   | 37
random | 5 |   | 5
reciprocal | 10 |   | 10
relu | 62 |   | 62
relu6 | 52 |   | 52
remainder | 33 |   | 33
round | 10 |   | 10
rsqrt | 10 |   | 10
sgn | 5 |   | 5
sigmoid | 10 |   | 10
sign | 10 |   | 10
sin | 10 |   | 10
sinh | 5 |   | 5
Softmax | 24 |   | 24
Softmax2d | 10 |   | 10
split | 11 |   | 11
sqrt | 10 |   | 10
square | 10 |   | 10
stack | 48 |   | 48
sum | 32 |   | 32
tan | 10 |   | 10
tanh | 10 |   | 10
trunc | 10 |   | 10
uniform | 5 |   | 5
unique | 5 |   | 5
weight_norm_sparsifier_step | 291 |   | 291
zero | 5 |   | 5
Total Cases | 4284 | 689 | 4973



</body>

</html>

**Comment by DiweiSun** [comment]
<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/diweisun/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/diweisun/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<body link="#467886" vlink="#96607D">


Conv   with Eager mode, warm up=30, iterations for benchmark=100 | Round1 | Round2 | Round3 | Round4 | Round5 | Round6 | Round7
-- | -- | -- | -- | -- | -- | -- | --
Conv1d_IC256_OC256_kernel3_stride2_N4_L64_cpu(us) | 71.44 | 71.287 | 72.682 | 71.586 | 70.899 | 70.483 | 71.467
Conv1d_IC256_OC256_kernel3_stride2_N4_L64_cpu(us) | 195.821 | 193.493 | 191.038 | 193.897 | 187.042 | 190.689 | 192.259
ConvTranspose1d_IC2016_OC1026_kernel1024_stride256_N1_L224_cpu(us) | 8064312.2 | 8053338.8 | 8026532.5 | 7926867.7 | 7970893.1 | 7929371.9 | 7902148.9
ConvTranspose1d_IC128_OC256_kernel3_stride1_N1_L64_cpu(us) | 143.425 | 151.248 | 150.339 | 154.585 | 154.388 | 147.815 | 148.945
ConvTranspose1d_IC256_OC256_kernel3_stride2_N4_L64_cpu(us) | 234.389 | 244.115 | 236.081 | 248.788 | 256.239 | 230.393 | 230.862
Conv2d_IC256_OC256_kernel3_stride1_N1_H16_W16_G1_pad0_cpu(us) | 266.707 | 269.292 | 269.272 | 266.867 | 276.622 | 264.835 | 274.649
ConvTranspose2d_IC256_OC256_kernel3_stride1_N1_H16_W16_G1_pad0_cpu(us) | 304.552 | 301.545 | 298.032 | 310.901 | 310.48 | 299.24 | 308.035
Conv2dPointwise_IC256_OC256_stride1_N1_H16_W16_G1_pad0_cpu(us) | 167.908 | 169.647 | 171.451 | 171.564 | 173.664 | 171.053 | 172.877
Conv3d_IC64_OC64_kernel3_stride1_N8_D4_H16_W16_cpu(us) | 326.112 | 327.466 | 328.105 | 331.639 | 326.694 | 327.777 | 333.634
ConvTranspose3d_IC64_OC64_kernel3_stride1_N8_D4_H16_W16_cpu(us) | 1477.432 | 1538.531 | 1618.35 | 1427.353 | 1678.129 | 1582.357 | 1611.974



</body>

</html>

**Review by chuanqi129 — APPROVED** [review]

**Comment by chuanqi129** [comment]
Suggest to rename the PR as "Add 1 test case for Convtranspose1D in op microbenchmark"

**Review by jgong5 — APPROVED** [review]

**Review by atalman — APPROVED** [review]
